draft-private-access-tokens.md

```diff
@@ -727,9 +727,41 @@ signal, such as an IP address, especially since malicious clients can generate g
 Private Access Tokens and for Origins to work. However, similar DoS vectors already exist
 for Origins, e.g., at the underlying TLS layer.
 
+## Man-in-the-Middle Attacks
+
+An attacker that can act as a man-in-the-middle for Mediator/Issuer communication can
+influence or disrupt the ability for the Issuer to correctly rate-limit token issuance.
+Communication between Mediators and Issuers MUST use a secure HTTPS connection, and SHOULD
+prevent man-in-the-middle attacks by employing an approach like TLS certificate pinning.
+
+An attacker than can act as a man-in-the-middle for Client/Origin communication can
+observe a TokenChallenge, and can view a Token being presented for authentication
+to an Origin. Origins can avoid this Token being used for some future connection
+by the attacker impersonating the Client by ensuring that the redemption_nonce
+presented in the TokenChallenge is bound to the specific TLS session with the
+Client.
```

Comment:
```suggestion
to an Origin. The protocol in this document does not allow Origins to protect against this type of attack.

[OPEN ISSUE: unless both client and server independently derive the challenge from the underlying channel, this problem will always remain.]
```

Reply:
@chris-wood hm, I'd like to have a bit of a better answer here rather than declaring we can't protect against it.

The main point, from the issue, was not to prevent the client-origin communication from having an intermediary watching things, but to stop the intermediary from harvesting tokens that it could then use in _other_ connections/sessions/contexts as its own client.

If the origin can require that the challenge/nonce is specific to a given session, can't it prove that this wasn't a token harvested on behalf of an unrelated connection?

Reply:
> If the origin can require that the challenge/nonce is specific to a given session, can't it prove that this wasn't a token harvested on behalf of an unrelated connection?

When the connection is MITM'd, the session is with the attacker, not the client, so this just doesn't seem to work. A workaround would be to derive the nonce _from_ the connection, a la via [the Export API](https://datatracker.ietf.org/doc/html/rfc8446#section-7.5), but that's pretty uncommon. Things like WebAuthn have [slots for this type of binding](https://www.w3.org/TR/webauthn-2/#dom-collectedclientdata-tokenbinding) but it's basically never used in practice.

Reply:
Right, the session is with the attacker, and then the client behind the attacker. Ah, so you’re saying the attacker could have other clients that it’s trying to squash together into one session?

Reply:
Do I understand correctly that this is a machine-on-the-side attacker, not a MITM attacker? AIUI, the attacker:
- watches the challenge and the token go by, harvesting both; and
- later spoofs a client to reach the origin and uses a token that matches the origin's challenge.

If my understanding is correct, then the crux of the issue is the risk that the nonce might repeat (which makes it a bad nonce.)

If so, it seems to me that @tfpauly's text suggests that an Origin can make a strong nonce by deriving it from something unique to the session as seen by the Origin (for example: timestamp + H(session key) + H(some client info)). I could of course be misunderstanding the issue and the text :-)

Reply:
@tfpauly I updated my suggestion to try and make the limitation (and open problem) more clear.

Reply:
@janaiyengar the attacker needs to intercept TLS to see the challenge and resulting Token, so it's not enough to be on the side, but maybe I misunderstood you?

Reply:
Does the extra ability to present multiple tokens in the same session, versus the attacker just mitming the individual connections and intercepting/replaying the tokens separately there introduce much of an issue? Though you can sort of solve both with the exporter secret solution from the TLS session.